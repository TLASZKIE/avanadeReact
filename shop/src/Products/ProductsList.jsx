import React from 'react'
import PropTypes from 'prop-types'
import ProductsPanel from './ProductsPanel'

const ProductsList = props => {
  const { onBuy } = props
  const specialProduct = props.products.filter(p => p.specialOffer)
  const normalProduct = props.products.filter(p => !p.specialOffer)

  return (
    <div>
      <ProductsPanel
        products={specialProduct}
        title="Special offer"
        onBuy={onBuy}
      />
      <ProductsPanel products={normalProduct} title="Products" onBuy={onBuy} />
    </div>
  )
}

ProductsList.propTypes = {}

export default ProductsList
