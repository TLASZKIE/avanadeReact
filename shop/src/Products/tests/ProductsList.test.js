import React from 'react'
import ReactDOM from 'react-dom'
import ProductsList from '../ProductsList'
import renderer from 'react-test-renderer'

//4/ The most primitive smoke test.
it('renders without crashing', () => {
  const div = document.createElement('div')

  const products = [{ name: 'name', id: 1, isSpecial: true, price: 1600 }]
  ReactDOM.render(<ProductsList products={products} />, div)
})

it('renders correctly', () => {
  const products = [{ name: 'name', id: 1, isSpecial: true, price: 1600 }]
  const tree = renderer.create(<ProductsList products={products} />).toJSON()

  expect(tree).toMatchSnapshot()
})
