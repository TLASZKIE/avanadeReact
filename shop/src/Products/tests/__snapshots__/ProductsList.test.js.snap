// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="products-panel"
  >
    <h1>
      Special offer
    </h1>
    <div
      className="products-product special"
    >
      <img
        alt="name"
        height="100"
        onClick={[Function]}
        src={undefined}
        width="100"
      />
      <div>
        name
      </div>
      <div>
        1600
      </div>
    </div>
  </div>
  <div
    className="products-panel"
  >
    <h1>
      Products
    </h1>
  </div>
</div>
`;
